### PR TITLE
[node]: add props to repl.Context

### PR DESCRIPTION
**Pull request by islishude**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://nodejs.org/dist/latest-v10.x/docs/api/repl.html#repl_global_and_local_scope>>
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

ref #30535

### CODE REVIEW

**Review by ZaneHannanAU — APPROVED** [review]

**Review by r3nya — APPROVED** [review]

**Review by BrunoScheufler — APPROVED** [review]

**Review by Flarna — APPROVED** [review]

File: types/node/v9/index.d.ts [hunk 1]
```diff
@@ -2390,7 +2390,9 @@ declare module "readline" {
 }
 
 declare module "vm" {
-    export interface Context { }
+    export interface Context {
```
Comment (Flarna) [inline types/node/v9/index.d.ts]: There seems to be a trailing whitespace according to (failing) CI

File: types/node/index.d.ts [hunk 2]
```diff
@@ -2501,7 +2501,9 @@ declare module "readline" {
 }
 
 declare module "vm" {
-    interface Context { }
+    interface Context {
```
Comment (Flarna) [inline types/node/index.d.ts]: There seems to be a trailing whitespace according to CI

File: types/node/node-tests.ts [hunk 3]
```diff
@@ -3675,6 +3675,12 @@ import * as p from "process";
         function test() {
             throw new repl.Recoverable(new Error("test"));
         }
+
+        _server.context.key0 = 1;
```
Comment (Flarna) [inline types/node/node-tests.ts]: In typescript 2.1 it's needed to write `_server.context['key0'] = 1;`

**Comment by isLishude** [comment]
It seems that testing failure is not a problem with my PR.Some libs are using `<>` instead of  keyword `as`  for type assertion.

image

**Comment by Flarna** [comment]
yes, fails are unrelated. There are several PRs around fixing at least some of them (e.g. #30691).

**Comment by isLishude** [comment]
Now the error is `The job exceeded the maximum time limit for jobs, and has been terminated.`

How to fix it ?

**Pull request merged**